protocol/contracts/voting/MasterCandidate.sol

```diff
@@ -0,0 +1,191 @@
+import "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import "@openzeppelin/contracts/utils/cryptography/draft-EIP712.sol";
+
+/**
+ * @title an NFT for each master candidate, with associated endorsements
+ * 
+ * @notice Captures key aspects of temple governance on chain. Namely
+ * 
+ *   - Anyone in temple discord with at least the Templar Role + verified
+ *     wallet can nominate themselves for candidacy.
+ *   - Any user can endorse candidates
+ *   - EIP712 structured data signing to allow 'gasless voting' via a relay
+ * 
+ * Which masters are currently voted in/active in the multisig etc is handled
+ * at the social layer.
+ */
+contract MasterCandidate is ERC721, Ownable, EIP712 {
+    using Strings for uint256;
+    using Counters for Counters.Counter;
+
+    bytes32 public immutable TOGGLE_ENDORSEMENTS_FOR_TYPEHASH = keccak256("toggleEndorsementsFor(address account, uint256[] discordIds, uint256 deadline, uint256 nonce)");
+    mapping(address => Counters.Counter) public _nonces;
+
+    /// @notice endorsements per nomination
+    mapping(address => mapping (uint256 => bool)) public endorsementsBy;
+
+    /// @dev merkle tree root of (address, claimShare)
+    bytes32 public merkleRoot;
+
+    /// @notice Token which nomination fee is charged in
+    IERC20 public immutable nominationFeeToken;
+
+    /// @notice Nomination fee amount (in feeToken)
+    uint256 public nominationFee;
+
+    /// @notice metadata per nomination
+    mapping(uint256 => Candidate) public candidates;
+
+    struct Candidate {
+        /// @notice candidate discord id
+        uint256 discordId;
+
+        /// @notice Discord handle (static, so if changed in discord needs to be manually updated)
+        string discordHandle;
+
+        /// @notice twitter handle (static, so if changed, needs to be manually updated)
+        string twitterHandle;
+    }
+
+    constructor(
+        IERC20 _nominationFeeToken,
+        uint256 _nominationFee
+    ) ERC721("Temple Master Candidate", "MASTER CANDIDATE") EIP712(name(), "1") {
+        nominationFeeToken = _nominationFeeToken;
+        nominationFee = _nominationFee;
+    }
+
+    /**
+     * @dev Change the nomination fee
+     * 
+     * Designed to be low, but a defense in depth to stop peeps putting up a nomination
+     * for the lolz.
+     */
+    function setNominationFee(uint256 _nominationFee) external onlyOwner {
+        nominationFee = _nominationFee;
+    }
+
+    /**
+     * @dev periodically update the merkle tree root to match discord membership.
+     * NOTE: If a user has already minted their 'master candidate' NFT, this doesn't
+     * remove it for them.
+     * 
+     * The voting UI only shows candidates in the intersection of the set
+     * MasterCandidate NFT and Discord Users.
+     */
+    function updateMerkleRoot(bytes32 _merkleRoot) external onlyOwner {
+        merkleRoot = _merkleRoot;
+    }
+
+    function nominate(
```

Comment:
Anyone in the temple discord can nominate themselves (for a small fee to stop spamoors)